Add Chi routes, document the new structure

ramperher opened the pull request.

ezr-ondrej commented on webserver/webserver.go:
```diff
@@ -15,47 +15,23 @@ limitations under the License.
 
 package main
```
For the reasons above, I'd move this file into `cmd/webserver.go`

ramperher replied:
Done

ezr-ondrej commented on routes/routes.go:
```diff
@@ -0,0 +1,76 @@
+		rw.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(rw).Encode(Response{Status: "error found: " + err.Error()})
+	} else {
+		fmt.Println("Received: {\"name\": \"" + userData.Name + "\", \"msg\": \"" + userData.Message + "\"}")
```
This is correct, but idiomatic go would probably prefer a string formatting, when there is more string concatenation.
```suggestion
		fmt.Printf("Received: {\"name\": \"%s\", \"msg\": \"%s\"}\n", userData.Name, userData.Message)
```
OR
```suggestion
		fmt.Println(fmt.Sprintf("Received: {\"name\": \"%s\", \"msg\": \"%s\"}\n", userData.Name, userData.Message))
```

Note most of the print functions are using the variadic functions that go offers: https://gobyexample.com/variadic-functions

ramperher replied:
That looks better for sure! I've included your second suggestion.

ezr-ondrej reviewed (commented):
You've very correctly captured the gist of what chi offers and how it should work. I think the use of encoders and decoders is done pretty well :orange_heart:
I like the choice of routes package and separate setup for router and endpoints.
I think having the handler in the routes package directly in this simple case is also correct.

I've got a question to the project layout: Why did you chose to separate this into two distinct go projects?

An usual way in go is to have main (executable) files in `/cmd` and the project defined in a root directory, there is an "standard go project structure" defined in https://github.com/golang-standards/project-layout it's no dogma and what you did is not incorrect, there might be usecases for having two projects in one github repo if you'd want to import them separatelly in other external projects.

Having the project in root and having only single project in the repo makes it easy to use github as source and your modules are local modules  and are more easily manged together.
Your project then becomes "github.com/rampherher/dummy-webserver"

ramperher commented:
Thanks @ezr-ondrej for your review! I've addressed your feedback, also answering your open questions below:

> I've got a question to the project layout: Why did you chose to separate this into two distinct go projects?

Just trying the creation of multiple modules at the same time, but finally I decided to follow the standard project structure to be more coherent.

ezr-ondrej reviewed (approved):
I believe this is great step :)
IMO next step would be to add some tests, lets talk about it online :+1:
